app/customer.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from app.car import Car
+from app.point import Point
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(
+            self,
+            name: str,
+            product_cart: dict[str, int],
+            location: Point,
+            money: float,
+            car: Car
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart or {}
+        self.location = location
+        self.money = money
+        self.car = car
+
+    @classmethod
+    def create_customer(
+            cls,
+            source: dict[str, str | dict[str, int] | list[int] | int]
+    ) -> Customer:
+        return cls(
+            source["name"],
+            source["product_cart"],
+            Point.create_from_location(source["location"]),
+            source["money"],
+            Car.create_car(source["car"])
```

Comment:
To prevent `KeyError` exceptions, consider using the `get` method with default values when accessing dictionary keys, such as `source.get('name', 'default_name')`, `source.get('product_cart', {})`, `source.get('location', [0, 0])`, `source.get('money', 0.0)`, and `source.get('car', {})`. This will make the code more robust against missing keys.